packages/contracts/contracts/minter-suite/Minters/MinterSetPriceERC20V4.sol

```diff
@@ -337,12 +364,35 @@ contract MinterSetPriceERC20V4 is
     function purchase_H4M(
         uint256 _projectId
     ) external payable returns (uint256 tokenId) {
-        tokenId = purchaseTo_do6(msg.sender, _projectId);
+        // pass max price as msg.value, currency address as ETH
+        tokenId = purchaseTo_do6({
+            _to: msg.sender,
+            _projectId: _projectId,
+            _maxPricePerToken: msg.value,
+            _currencyAddress: address(0)
+        });
         return tokenId;
     }
 
     /**
-     * @notice Purchases a token from project `_projectId` and sets
+     * @notice gas-optimized version of purchase(uint256,uint256,address).
+     */
+    function purchase_H4M(
```

Comment:
nit: this would be `purchase_a1A`, but I would be fine just removing the gas-optimized version in this update.
ref: https://emn178.github.io/solidity-optimize-name/

Reply:
Ah, thanks for sharing! Removed this 🙏